pkg/controller/daemon/daemoncontroller.go

```diff
@@ -765,14 +784,14 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 	if dsNeedsSync && ds.DeletionTimestamp == nil {
 		switch ds.Spec.UpdateStrategy.Type {
 		case extensions.RollingUpdateDaemonSetStrategyType:
-			err = dsc.rollingUpdate(ds)
+			err = dsc.rollingUpdate(ds, nodeToDaemonPods)
```

Comment:
In other controllers, we expect changes we make to be observed only in the next sync, and we re-queue when appropriate. Does DaemonSet not work that way?